pyccel/ast/core.py

```diff
@@ -2113,7 +2116,7 @@ def __new__(
         class_type = cls_base \
             or dtype.__class__.__name__.startswith('Pyccel')
         alloweddtypes = (NativeBool, NativeRange, NativeString,
```

Comment:
I have just realized that `NativeBool` is checked for twice in the `if/elif` block below, so maybe we should remove it from the tuple above?

As a side note, I think the `if/elif` block below should be improved: it is more natural to raise a TypeError after an `else` statement, making it clear that we are covering all possible cases (by looking at the block as it is, one is left with some doubts about it).

Reply:
How do you suggest clarifying that? With an `elif:pass` ?

Reply:
Yes, I would write a `pass` and maybe a line of comment above it.